packages/medusa/src/strategies/product/import.ts

```diff
@@ -0,0 +1,772 @@
+   */
+  private async createProducts(
+    batchJobId: string,
+    transactionManager: EntityManager
+  ): Promise<void> {
+    const productOps = await this.getImportDataFromRedis(
+      batchJobId,
+      OperationType.ProductCreate
+    )
+
+    for (const productOp of productOps) {
+      try {
+        await this.productService_
+          .withTransaction(transactionManager)
+          .create(
+            transformProductData(productOp) as unknown as CreateProductInput
+          )
+      } catch (e) {
+        ProductImportStrategy.throwDescriptiveError(productOp)
+      }
+
+      this.updateProgress_(batchJobId)
+    }
+  }
+
+  /**
+   * Method updates existing products in the DB using a CSV row data.
+   *
+   * @param batchJobId - An id of the current batch job being processed.
+   * @param transactionManager - Transaction manager responsible for current batch import.
+   */
+  private async updateProducts(
+    batchJobId: string,
+    transactionManager: EntityManager
+  ): Promise<void> {
+    const productOps = await this.getImportDataFromRedis(
+      batchJobId,
+      OperationType.ProductUpdate
+    )
+
+    for (const productOp of productOps) {
+      try {
+        await this.productService_
+          .withTransaction(transactionManager)
+          .update(
+            productOp["product.id"] as string,
+            transformProductData(productOp)
+          )
+      } catch (e) {
+        ProductImportStrategy.throwDescriptiveError(productOp)
+      }
+
+      this.updateProgress_(batchJobId)
+    }
+  }
+
+  /**
+   * Method creates product variants from a CSV data.
+   * Method also handles processing of variant options.
+   *
+   * @param batchJobId - An id of the current batch job being processed.
+   * @param transactionManager - Transaction manager responsible for current batch import.
+   */
+  private async createVariants(
+    batchJobId: string,
+    transactionManager: EntityManager
+  ): Promise<void> {
+    const productRepo = transactionManager.getCustomRepository(
+      this.productRepo_
+    )
+
+    const variantOps = await this.getImportDataFromRedis(
+      batchJobId,
+      OperationType.VariantCreate
+    )
+
+    for (const variantOp of variantOps) {
+      try {
+        const variant = transformVariantData(variantOp)
+
+        const product = await productRepo.findOne({
+          where: { handle: variantOp["product.handle"] },
+          relations: ["variants", "variants.options", "options"],
+        })
+
+        const optionIds =
+          (variantOp["product.options"] as Record<string, string>[])?.map(
+            (variantOption) =>
+              product!.options.find(
+                (createdProductOption) =>
+                  createdProductOption.title === variantOption.title
+              )!.id
+          ) || []
+
+        variant.options =
+          (variant.options as object[])?.map((o, index) => ({
```

Comment:
suggestion: we should avoid typing as object and privilege Record instead, can't we type the result of the transformation?